### PR TITLE
Feature/#17 dashboard

### LINKED ISSUE #17: Dashboard view
Overview dashboard view.
# View
- Three fields with statistics
- Button to start sessions. Link to find users view
- Navigation element for sections
  - Dashboard
  - Users (new user)
  - Sessions (startedSessions)

### PR DESCRIPTION
**Pull request by Crevil**
closes #17

### CODE REVIEW

File: app/base-view/base-view.component.html [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    class="background"
+    (loaded)="startBackgroundAnimation()"></GridLayout>
+
+  <StackLayout #mainContainer class="main-container">
+    <ContentView [visibility]="hideNavigation ? 'collapsed' : 'visible'">
+      <rom-tabs></rom-tabs>
+    </ContentView>
+    <ng-content></ng-content>
+  </StackLayout>
+</GridLayout>
```
Comment (Karnich) [inline app/base-view/base-view.component.html]: oh crap. need newline

File: app/base-view/base-view.component.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+  private showMainContent() {
+    const mainContainer = this.mainContainer.nativeElement as View;
+    if (!this.animate) {
+      mainContainer.opacity = 1;
+      return;
+    }
+
+    const animations = [];
+
```
Comment (Karnich) [inline app/base-view/base-view.component.ts]: alot of linesspaces used for nothing? maybe remove 1 or 2?

File: app/base-view/tabs/tab/tab.component.html [hunk 3]
```diff
@@ -0,0 +1,4 @@
+<StackLayout class="tab" [nsRouterLink]="[route]">
+	<Label class="text" [text]="text"></Label>
+	<StackLayout #line class="line" scaleX="0.01" scaleY="1.0"></StackLayout>
+</StackLayout>
```
Comment (Karnich) [inline app/base-view/tabs/tab/tab.component.html]: newline

File: app/base-view/tabs/tabs.component.html [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+	<ContentView col="0">
+		<rom-tab text="Dashboard" route="/dashboard"></rom-tab>
+	</ContentView>
+	<ContentView col="1">
+		<rom-tab text="User" route="/newUser"></rom-tab>
+	</ContentView>
+	<ContentView col="2">
+		<rom-tab text="Sessions" route="/startedSessions"></rom-tab>
+	</ContentView>
+</GridLayout>
```
Comment (Karnich) [inline app/base-view/tabs/tabs.component.html]: newline

File: app/bluetooth-details/bluetooth-details.component.ts [hunk 5]
```diff
@@ -30,5 +27,6 @@ export class BluetoothDetailsComponent implements OnInit {
   public start(): void {
     this.sessionService.startCurrentSession();
     // TODO: Route someplace?
```
Comment (Karnich) [inline app/bluetooth-details/bluetooth-details.component.ts]: remove todo comment when you added routing

File: app/bluetooth/bluetooth.component.html [hunk 6]
```diff
@@ -1,23 +1,18 @@
...
-      </Button>
-		</GridLayout>
-    
-	</StackLayout>
+		<Button row="2" class="scan-btn" (tap)="scan()" [isEnabled]="!isScanning" [text]="isScanning ? 'Scanning' : 'Scan'"></Button>
+	</GridLayout>
 
-</GridLayout>
\ No newline at end of file
+</rom-base-view>
```
Comment (Karnich) [inline app/bluetooth/bluetooth.component.html]: newline

File: app/bluetooth/bluetooth.component.html [hunk 7]
```diff
@@ -1,23 +1,18 @@
...
 
-	<GridLayout #background class="background"></GridLayout>
+	<GridLayout rows="auto, *, auto" class="ble-grid">
+		<Label row="0" class="main-label" text="Select a racket"></Label>
 
-	<StackLayout #mainContainer class="main-container">
+		<ListView row="1" class="racket-list" [items]="peripherals" (itemTap)="onPeripheralTap($event)">
+			<template let-item="item">
+				<StackLayout class="racket-item">
+					<Label [text]='item.name'></Label>
```
Comment (Karnich) [inline app/bluetooth/bluetooth.component.html]: this file have some crazy indent/spacing going on

File: app/bluetooth/bluetooth.component.ts [hunk 8]
```diff
@@ -34,20 +34,20 @@ export class BluetoothComponent implements OnInit {
...
+  public scan(): Promise<void> {
     this.isScanning = true;
     this.peripherals.length = 0;
-    this.bluetoothService.isBluetoothEnabled().then( (enabled) => {
+    return this.bluetoothService.isBluetoothEnabled().then((enabled) => {
       if (!enabled) {
-        this.dialogService.alert("bluetooth is not enabled", "Bluetooth");
-        return;
+        return this.dialogService.alert("Bluetooth is not enabled")
+          .then(() => this.router.navigate(["dashboard"]));
```
Comment (Karnich) [inline app/bluetooth/bluetooth.component.ts]: you dont need to reroute because BLE is not enabled?

File: app/dashboard/dashboard.component.html [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+
+		<StackLayout row="2" class="buttons" orientation="horizontal">
+			<Button text="Log out" (tap)="goTo('login')"></Button>
+			<Button text="New user" (tap)="goTo('newUser')"></Button>
+			<Button text="Started sessions" (tap)="goTo('startedSessions')"></Button>
+		</StackLayout>
+
+	</GridLayout>
+
+</rom-base-view>
```
Comment (Karnich) [inline app/dashboard/dashboard.component.html]: newline

File: app/dashboard/tile/tile.component.html [hunk 10]
```diff
@@ -0,0 +1,4 @@
+<StackLayout class="frame">
+	<Label class="value" [text]="value"></Label>
+	<Label class="text" [text]="text"></Label>
+</StackLayout>
```
Comment (Karnich) [inline app/dashboard/tile/tile.component.html:4]: newline

File: app/login/login.component.html [hunk 11]
```diff
@@ -1,67 +1,61 @@
...
       translateY="50">
       <Label
         [text]="isLoggingIn ? 'Sign up here' : 'Back to login'"></Label>
     </StackLayout>
-  </StackLayout>
-
-</GridLayout>
\ No newline at end of file
+  </GridLayout>
+</rom-base-view>
```
Comment (Karnich) [inline app/login/login.component.html]: newline

File: app/started-sessions/started-sessions.component.html [hunk 12]
```diff
@@ -33,5 +28,4 @@
     </ListView>
 
   </GridLayout>
-
-</GridLayout>
+</rom-base-view>
```
Comment (Karnich) [inline app/started-sessions/started-sessions.component.html]: newline

**Review by Karnich — CHANGES_REQUESTED** [review]

File: app/bluetooth/bluetooth.component.ts (hunk 8 above)
Comment (Crevil) [reply app/bluetooth/bluetooth.component.ts]: Left over for demo flow. Removing.

**Comment by Crevil** [comment]
@Karnich I've made the changes you requested. Good catches.

**Pull request merged**